pkg/quota/image/imagestreamtag_evaluator.go

```diff
@@ -0,0 +1,106 @@
+	"k8s.io/kubernetes/pkg/quota/generic"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const imageStreamTagEvaluatorName = "Evaluator.ImageStreamTag"
+
+// NewImageStreamTagEvaluator computes resource usage of ImageStreamsTags. Its sole purpose is to handle
+// UPDATE admission operations on imageStreamTags resource.
+func NewImageStreamTagEvaluator(istNamespacer osclient.ImageStreamTagsNamespacer, isNamespacer osclient.ImageStreamsNamespacer) kquota.Evaluator {
+	computeResources := []kapi.ResourceName{
+		imageapi.ResourceImageStreams,
+	}
+
+	matchesScopeFunc := func(kapi.ResourceQuotaScope, runtime.Object) bool { return true }
+	getFuncByNamespace := func(namespace, id string) (runtime.Object, error) {
+		isName, tag, err := imageapi.ParseImageStreamTagName(id)
+		if err != nil {
+			return nil, err
+		}
+
+		obj, err := istNamespacer.ImageStreamTags(namespace).Get(isName, tag)
+		if err != nil {
+			if !kerrors.IsNotFound(err) {
+				return nil, err
+			}
+			obj = &imageapi.ImageStreamTag{
+				ObjectMeta: kapi.ObjectMeta{
+					Namespace: namespace,
+					Name:      id,
+				},
+			}
+		}
+		return obj, nil
+	}
+
+	return &generic.GenericEvaluator{
+		Name:                       imageStreamTagEvaluatorName,
+		InternalGroupKind:          imageapi.Kind("ImageStreamTag"),
+		InternalOperationResources: map[admission.Operation][]kapi.ResourceName{admission.Update: computeResources},
+		MatchedResourceNames:       computeResources,
+		MatchesScopeFunc:           matchesScopeFunc,
+		UsageFunc:                  makeImageStreamTagAdmissionUsageFunc(isNamespacer),
+		GetFuncByNamespace:         getFuncByNamespace,
+		ListFuncByNamespace: func(namespace string, options kapi.ListOptions) (runtime.Object, error) {
+			return &imageapi.ImageStreamTagList{}, nil
+		},
+		ConstraintsFunc: imageStreamTagConstraintsFunc,
+	}
+}
+
+// imageStreamTagConstraintsFunc checks that given object is an image stream tag
+func imageStreamTagConstraintsFunc(required []kapi.ResourceName, object runtime.Object) error {
+	if _, ok := object.(*imageapi.ImageStreamTag); !ok {
+		return fmt.Errorf("unexpected input object %v", object)
+	}
+	return nil
+}
+
+// makeImageStreamTagAdmissionUsageFunc returns a function that computes a resource usage for given image
+// stream tag during admission.
+func makeImageStreamTagAdmissionUsageFunc(isNamespacer osclient.ImageStreamsNamespacer) generic.UsageFunc {
+	return func(object runtime.Object) kapi.ResourceList {
+		ist, ok := object.(*imageapi.ImageStreamTag)
+		if !ok {
+			return kapi.ResourceList{}
+		}
+
+		res := map[kapi.ResourceName]resource.Quantity{
+			imageapi.ResourceImageStreams: *resource.NewQuantity(0, resource.BinarySI),
+		}
+
+		isName, _, err := imageapi.ParseImageStreamTagName(ist.Name)
+		if err != nil {
+			glog.Error(err.Error())
```

Comment:
I woudn't clutter the error message with additional information. The error reported should be sufficient (see the [possible errors](https://github.com/openshift/origin/pull/8195/files#diff-78bcdfab573a18c6ae2ac4e175b48addR73)). The line is wrong nevertheless. The `err` needs to be handled with `utilruntime.HandleError`.